### DISCUSSION TITLE
uncompressed save?

### DISCUSSION THREAD

**Discussion by nschloe** [zarr-developers/zarr-python#1332] [Q&A] [answered]
With 

```python
import numpy as np
import zarr

data = np.random.rand(1_000_000)
zarr.save("out.zarr", data)
```

one can save a numpy array with the default compressor. How do I turn it off? This

```python
zarr.save("out.zarr", data, compressor=None)
```

only gives

```
TypeError: create() missing 1 required positional argument: 'shape'
```

**Comment by d-v-b** [comment, marked as answer]
Hi @nschloe, it looks like [`zarr.save`](https://github.com/zarr-developers/zarr-python/blob/0bf0b3b3444dd57debc8a6b5eacb6eb5c082c668/zarr/convenience.py#L261) doesn't forward array creation kwargs like `compressor` to the right function -- instead, `zarr.save` assumes that **kwargs is a dict of arrays? 

Personally, I don't like this function and I recommend instead using [`zarr.save_array`](https://github.com/zarr-developers/zarr-python/blob/0bf0b3b3444dd57debc8a6b5eacb6eb5c082c668/zarr/convenience.py#L34)

```python
 zarr.save_array("out.zarr", data, compressor=None)
```

**Comment by d-v-b** [comment]
We should probably remove or refactor any routine like `zarr.save` that offers incomplete access to the array creation API.

**Reply by jakirkham** [reply]
Agreed. There are several convenience APIs that have different levels of functionality. Trimming them down to one preferred method would be a net improvement for user experience and maintainability

**Reply by nschloe** [reply]
Would have definitely saved me some time.
